src/app/charts/chart.component.ts

```diff
@@ -76,24 +77,44 @@ export class ChartComponent implements OnChanges {
                 private indicatorService: IndicatorService,
                 private dataExportService: DataExportService,
                 private imageExportService: ImageExportService,
-                private authService: AuthService) {}
+                private authService: AuthService,
+                private changeDetector: ChangeDetectorRef) {}
+
+    ngAfterViewInit() {
+        // Manually trigger change detection in parent to avoid
+        // ExpressionChangedAfterItHasBeenCheckedError when extra params form
+        // calls to set initial values in ngAfterViewInit.
+        this.changeDetector.detectChanges();
+    }
 
-    ngOnChanges() {
+    ngOnChanges($event) {
         if (!this.scenario || !this.city || !this.models) { return; }
+        this.updateChart($event);
+    }
+
+    updateChart(extraParams: any) {
         this.chartData = [];
         this.rawChartData = [];
+
+        let params = {
+            climateModels: this.models,
+            unit: this.unit || this.chart.indicator.default_units,
+            // As a temporary solution, the time agg defaults to the 1st valid option.
```

Comment:
Hmm. We haven't addressed this. For now we should always be sending `time_aggregation: 'yearly'`, although its pretty likely that index 0 will always be 'yearly'. Definitely out of scope for this task, worth being a separate issue to address this since I think this happens in a few places.  We could also improve our handling of these by using the new string enums introduced in TypeScript 2.4 so that we wouldn't have to litter magic strings like 'yearly' throughout our code.